src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1195,6 +1223,100 @@ def construct_payload(self):
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())
+
+        has_Dockerfile = _has_dockerfile(self.get_argument_source(), dockerfile)
+        if not self.get_argument_source() or has_Dockerfile:
+            dockerfile_content = _get_dockerfile_content(self.get_argument_repo(), self.get_argument_branch(), token, self.get_argument_source(), self.get_argument_context_path(), dockerfile)
+            ingress, target_port = _get_ingress_and_target_port(self.get_argument_ingress(), self.get_argument_target_port(), dockerfile_content)
+
+        # Construct ContainerApp
+        resource_group = ResourceGroup(self.cmd, env_rg, location=location)
+        env = ContainerAppEnvironment(self.cmd, env_name, resource_group, location=location)
+        app = ContainerApp(self.cmd, self.get_argument_name(), resource_group, None, image, env, target_port, self.get_argument_registry_server(), self.get_argument_registry_user(), self.get_argument_registry_pass(), self.get_argument_env_vars(), self.get_argument_workload_profile_name(), ingress)
+
+        # Fetch registry credentials
+        _get_registry_details(self.cmd, app, self.get_argument_source())  # fetch ACR creds from arguments registry arguments
+
+        if self.get_argument_source():
+            # Uses buildpacks or an ACR Task to generate image if Dockerfile was not provided by the user
+            app.run_acr_build(dockerfile, self.get_argument_source(), quiet=False, build_from_source=not has_Dockerfile)
+            # Update image
+            if self.containerapp_def["properties"]["template"]["containers"] is None or len(self.containerapp_def["properties"]["template"]["containers"]) == 0:
+                raise ValidationError(
+                    "The container app '{}' does not have any containers. Please use --image to set the image for the container app".format(
+                        self.get_argument_name()))
+            self.containerapp_def["properties"]["template"]["containers"][0]["image"] = HELLO_WORLD_IMAGE if app.image is None else app.image
+
+        if self.get_argument_repo():
+            # Get GitHub access token
+            token = get_token(self.cmd, self.get_argument_repo(), self.get_argument_token())
+            _create_github_action(app, env, self.get_argument_service_principal_client_id(), self.get_argument_service_principal_client_secret(),
+                                  self.get_argument_service_principal_tenant_id(), self.get_argument_branch(), token, self.get_argument_repo(), self.get_argument_context_path())
+            cache_github_token(self.cmd, token, self.get_argument_repo())
+            r = self.client.show(cmd=self.cmd, resource_group_name=env_rg, name=app.name)
```

Comment:
What's the point for this `self.client.show`?
They just create github action, nothing for containerapp resource.

For `post_process`, we carry the response `r` to handle the scenario where `--identity` is `"system"`, which need to put containerapp twice (one in `create()`, the other in `post_process()`), return the second response.

So if there is no `put` or `patch `request for containerapp, no need to modify response.

Reply:
`self.client.show` here fetches the container app after the source control finished and the new image was deployed successfully ( so the initial image `mcr.microsoft.com/k8se/quickstart` will be replaced by the image created and deployed by source control)
It is required here to show the updated image in the response.